client/verta/verta/endpoint/resources.py

```diff
@@ -73,15 +80,26 @@ def _validate_memory(self, memory):
         if not re.match(r"^[0-9]+[e]?[0-9]*[E|P|T|G|M|K]?[i]?$", memory):
             raise ValueError(self.MEMORY_ERR_MSG)
 
+    def _validate_nvidia_gpu(self, nvidia_gpu):
+        if not isinstance(nvidia_gpu, NvidiaGPU):
+            raise TypeError(
+                "`nvidia_gpu` must be an instance of `verta.endpoint.NvidiaGpu`"
+            )
+        return nvidia_gpu
+
     def _as_dict(self):
         d = dict()
         if self.cpu is not None:
             d["cpu_millis"] = int(self.cpu * 1000)
         if self.memory is not None:
             d["memory"] = self.memory
+        if self.nvidia_gpu is not None:
+            d["nvidia_gpu"] = self.nvidia_gpu._as_dict()
 
         return d
 
     @classmethod
-    def _from_dict(cls, rule_dict):
-        return cls(**rule_dict)
+    def _from_dict(cls, resources_dict):
+        if "nvidia_gpu" in resources_dict:
```

Comment:
We have to copy the dict if we're modifying its items. Otherwise it may affect the calling code in surprising ways.

```suggestion
        resources_dict = resources_dict.copy()
        if "nvidia_gpu" in resources_dict:
```